### PR TITLE
feature(artifacts-test): support to test suse linux

### PR DESCRIPTION
**Pull request by amoskong**
ticket: https://trello.com/c/oxsOLpBg/3489-8277-it-is-now-possible-to-install-scylla-on-suse-linux-enterprise-server-using-the-standard-rpm-packages

Currently we only want to support one release for Suse Enterprise Linux.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
Note: One scylla fix needs to be backported to 4.5

- https://github.com/scylladb/scylla/issues/9045
- https://github.com/scylladb/scylla-tools-java/commit/4ef8049e0717a2ea7998a21ff8529ae293a37a6c

**Comment by amoskong** [comment]
Current status:

Takuya already fixed the _yaml.so dependency issue in scylla-tools-java.

Currently we need to enable Legacy and Python_2 modules for resolving the python2-PyYAML and java-1.8 dependency.
But it doesn't work all the time (it works sometimes), I'm working to address the problem.

```
scylla-test@artifacts-sles15-jenkins-db-node-ee0cda78-0-1:~> sudo zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool
Repository Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool not found.
```

**Comment by amoskong** [comment]
I had resolved the dependency issue in the PR, moving the PR to ready state.

Now the test failed for scylla issues:

- https://github.com/scylladb/scylla/issues/9172
- https://github.com/scylladb/seastar/issues/930  (posted a fix to seastar)

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2003,6 +2036,19 @@ def install_scylla(self, scylla_repo):
...
+            self.remoter.sudo('zypper install -y rsync tcpdump screen')
+            self.download_scylla_repo(scylla_repo)
+            # self.remoter.sudo('zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Pool')
+            # self.remoter.sudo('zypper mr -e Python_2_Module_x86_64:SLE-Module-Python2-15-SP3-Updates')
+            # self.remoter.sudo('zypper mr -e Legacy_Module_x86_64:SLE-Module-Legacy15-SP3-Updates')
+            # self.remoter.sudo('zypper mr -e Legacy_Module_x86_64:SLE-Module-Legacy15-SP3-Pool')
+            self.remoter.sudo('SUSEConnect --product sle-module-legacy/15.3/x86_64')
+            self.remoter.sudo('SUSEConnect --product sle-module-python2/15.3/x86_64')
+            self.remoter.sudo('zypper install -y python2-PyYAML', ignore_status=True)
+            self.remoter.sudo('zypper install -y python3-PyYAML', ignore_status=True)
```
Comment (roydahan) [inline sdcm/cluster.py:2049]: Is this part of the official documentation?

**Review by roydahan — COMMENTED** [review]
All in all looks good to me, but AFAIU it doesn't work yet, right?

Is the test synced with the official documentation?
Can you please link here the documentation?

File: sdcm/cluster.py [hunk 2]
```diff
@@ -2292,7 +2345,8 @@ def install_mgmt(self, scylla_mgmt_repo: str, package_url: Optional[str] = None)
 
         if self.distro.is_rhel_like:
             self.install_epel()
-            self.remoter.sudo("yum install python36-PyYAML -y", retry=3)
```
Comment (dkropachev) [inline sdcm/cluster.py:2295]: Why this one was removed ?

**Review by dkropachev — CHANGES_REQUESTED** [review]

**Review by enaydanov — CHANGES_REQUESTED** [review]
Please, add unit test for `Distro.is_sles` and `Distro.is_sles15` to `unit_tests/test_utils_distro.py`

```
test@sles15:~> cat /etc/os-release
NAME="SLES"
VERSION="15-SP3"
VERSION_ID="15.3"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP3"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp3"
DOCUMENTATION_URL="https://documentation.suse.com/"
test@sles15:~>
```

File: sdcm/cluster.py (hunk 1 above)
Comment (amoskong) [reply sdcm/cluster.py:2049]: We don't have install guide for Suse now.

File: sdcm/cluster.py (hunk 2 above)
Comment (amoskong) [reply sdcm/cluster.py:2295]: added back, thanks

**Comment by amoskong** [comment]
Rebased the PR to latest master and added unittest.

The test sill failed for scylla issue, I just pinged in the fix PR:
- test job: https://jenkins.scylladb.com/job/scylla-master/job/artifacts/job/artifacts-sles15-test/12/console
- seastar issue: https://github.com/scylladb/seastar/issues/930
- fix: https://github.com/scylladb/seastar/pull/931

We can consider to merge this PR.

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1981,6 +2008,10 @@ def update_repo_cache(self):
                 self.remoter.run('sudo yum clean all')
                 self.remoter.run('sudo rm -rf /var/cache/yum/')
                 self.remoter.run('sudo yum makecache', retry=3)
+            elif self.distro.is_sles:
+                self.remoter.run('sudo zypper clean all')
+                self.remoter.run('sudo rm -rf /var/cache/zypp/')
+                self.remoter.run('sudo zypper refresh', retry=3)
```
Comment (enaydanov) [inline sdcm/cluster.py:2014]: Please, use `self.runner.sudo()` here

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1997,6 +2028,8 @@ def upgrade_system(self):
                 self.remoter.run('sudo yum update -y --skip-broken --disablerepo=epel', retry=3)
             else:
                 self.remoter.run('sudo yum update -y --skip-broken', retry=3)
+        elif self.distro.is_sles:
+            self.remoter.run('sudo zypper update -y -l')
```
Comment (enaydanov) [inline sdcm/cluster.py:2032]: Please, use `self.runner.sudo()` here

**Review by enaydanov — CHANGES_REQUESTED** [review]
Except run/sudo stuff I'm OK with these changes.

File: sdcm/cluster.py (hunk 3 above)
Comment (amoskong) [reply sdcm/cluster.py]: fixed.

File: sdcm/cluster.py [hunk 5]
```diff
@@ -1855,6 +1863,13 @@ def download_scylla_repo(self, scylla_repo):
             self.remoter.run('sudo chmod 644 %s' % repo_path)
             result = self.remoter.run('cat %s' % repo_path, verbose=True)
             verify_scylla_repo_file(result.stdout, is_rhel_like=True)
+        elif self.distro.is_sles:
+            repo_path = '/etc/zypp/repos.d/scylla.repo'
+            self.remoter.run('sudo curl -o %s -L %s' % (repo_path, scylla_repo))
+            self.remoter.run('sudo chown root:root %s' % repo_path)
+            self.remoter.run('sudo chmod 644 %s' % repo_path)
```
[the remark this answers is not in the document]
Comment (amoskong) [reply sdcm/cluster.py]: fixed.

**Review by enaydanov — APPROVED** [review]

**Review by roydahan — APPROVED** [review]

**Comment by roydahan** [comment]
@amoskong can you please check with @lauranovich there is a matching documentation?

**Pull request merged**